crates/edr_evm/src/test_utils.rs

```diff
@@ -174,8 +174,11 @@ pub async fn run_full_block<
 >(
     url: String,
     block_number: u64,
-    header_overrides_constructor: impl FnOnce(&block::Header) -> HeaderOverrides<ChainSpecT::Hardfork>,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<(
+    RemoteBlock<ChainSpecT>,
+    ForkedBlockchain<ChainSpecT>,
+    IrregularState,
+)> {
```

Comment:
Suggestion: Create a named struct with members to self-document what each returned value means. E.g.:

```rs
struct ForkedStateAndBlockchain<ChainSpecT> {
  pub expected_block: RemoteBlock<ChainSpecT>,
  pub prior_blockchain: ForkedBlockchain<ChainSpecT>,
  pub prior_irregular_state: IrregularState,
}
```